modules/base/rendering/screenspaceinsetblackout.cpp

```diff
@@ -0,0 +1,951 @@
+                                                                const Spline::Side& side)
+{
+    std::shared_lock lock(cornerMutex);
+    std::pair<glm::vec2, glm::vec2> result;
+    switch (side) {
+    case Spline::Side::Top:
+        result = std::make_pair(topLeftCorner, topRightCorner);
+        break;
+    case Spline::Side::Right:
+        result = std::make_pair(topRightCorner, bottomRightCorner);
+        break;
+    case Spline::Side::Bottom:
+        result = std::make_pair(bottomRightCorner, bottomLeftCorner);
+        break;
+    case Spline::Side::Left:
+        result = std::make_pair(bottomLeftCorner, topLeftCorner);
+        break;
+    }
+    return result;
+}
+
+void ScreenSpaceInsetBlackout::BlackoutShape::setCornerData(const Spline::Side& side,
+                                            const glm::vec2 corner0,
+                                            const glm::vec2 corner1)
+{
+    std::unique_lock lock(cornerMutex);
+    switch (side) {
+    case Spline::Side::Top:
+        topLeftCorner = corner0;
+        topRightCorner = corner1;
+        break;
+    case Spline::Side::Right:
+        topRightCorner = corner0;
+        bottomRightCorner = corner1;
+        break;
+    case Spline::Side::Bottom:
+        bottomRightCorner = corner0;
+        bottomLeftCorner = corner1;
+        break;
+    case Spline::Side::Left:
+        bottomLeftCorner = corner0;
+        topLeftCorner = corner1;
+        break;
+    }
+}
+
+bool ScreenSpaceInsetBlackout::BlackoutShape::isTextureDirty() {
+    return ( dirtyTexture
+        || top.dirtyTexture
+        || right.dirtyTexture
+        || bottom.dirtyTexture
+        || left.dirtyTexture
+    );
+}
+
+void ScreenSpaceInsetBlackout::BlackoutShape::updateSplineAndGui() {
+    Spline* refs[4];
+    bool addedPoint[4];
+    bool removedPoint[4];
+
+    refs[0] = &top;
+    refs[1] = &right;
+    refs[2] = &bottom;
+    refs[3] = &left;
+
+    addedPoint[0] = top.pointFlagNew;
+    addedPoint[1] = right.pointFlagNew;
+    addedPoint[2] = bottom.pointFlagNew;
+    addedPoint[3] = left.pointFlagNew;
+
+    removedPoint[0] = top.pointFlagDelete;
+    removedPoint[1] = right.pointFlagDelete;
+    removedPoint[2] = bottom.pointFlagDelete;
+    removedPoint[3] = left.pointFlagDelete;
+
+    for (int i = 0; i < 4; ++i) {
+        bool updatePropertyTree = false;
+        if (addedPoint[i]) {
+            refs[i]->addPoint();
+            refs[i]->pointFlagNew = false;
+            updatePropertyTree = true;
+        }
+        if (removedPoint[i]) {
+            refs[i]->removePoint();
+            refs[i]->pointFlagDelete = false;
+            updatePropertyTree = true;
+        }
+
+        // Need to remove and add the PropertySubOwner as the GUI does not reflect
+        // changes otherwise
+        if (updatePropertyTree) {
+            removePropertySubOwner(refs[i]);
+            addPropertySubOwner(refs[i]);
+        }
+    }
```

Comment:
```suggestion
    std::array<Spline*, 4> refs = { &top, &right, &bottom, &left };

    for (Spline* spline : refs) {
        bool updatePropertyTree = false;
        if (spline->pointFlagNew) {
            spline->addPoint();
            spline->pointFlagNew = false;
            updatePropertyTree = true;
        }
        if (spline->pointFlagDelete) {
            spline->removePoint();
            spline->pointFlagDelete = false;
            updatePropertyTree = true;
        }

        // Need to remove and add the PropertySubOwner as the GUI does not reflect
        // changes otherwise
        if (updatePropertyTree) {
            removePropertySubOwner(spline);
            addPropertySubOwner(spline);
        }
    }
```

Less chance for copy-and-paste errors in the oredering of things